### PR TITLE
Validation plugin

### PR DESCRIPTION
**Pull request by nicky-johnstone**
Adds the validation plugin and CSS.

This is the plugin that's been used in the blog project and also in the MLA WWW responsive stuff. Thought that it should be included in qcode-ui now that there's another place where it will be used; MLA ERP.

### CODE REVIEW

File: js/jquery.validation.js [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+                        myWindow.document.write(response.responseText);
+                    }
+                });
+            }
+            
+            $(this.element).on('submit.validate', validate);
+        },
+        
+        showValidationMessage: function($element, message) {
+            // Show the validation message with the message as the content for the given element.
```
Comment (nicky-johnstone) [inline js/jquery.validation.js]: There's `showValidationMessage` and `showMessage`.

`showValidationMessage` displays the qtips on an element while `showMessage` displays the `alert`, `notify`, or `error` messages.

I feel that one (or maybe both?) should be renamed as there could be confusion with the names as they currently stand.

Some ideas:

`showValidationMessage`
  * `showQtip`
  * `showHint`
  * `showRecordItemMessage`

Any better names for either `showValidationMessage` or `showMessage`?

File: js/jquery.validation.js [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+            return $element;
+        },
+        
+        showMessage: function(type, message) {
+            // Show the message of the type given with message as the content.
+            if (! this[type]) {
+                // Message area doesn't exist so create it.
+                var messageDiv = $('<div></div>').addClass(this.options.messages[type].classes)
+                var messageContent = $('<span></span>').text(message).addClass('message-content');
+                var messageClose = $('<span></span>').addClass('icon icon--cross');
```
Comment (nicky-johnstone) [inline js/jquery.validation.js]: Not sure how we should deal with the close icon on messages. Should an icon font be included along with CSS for it?

Comment (bvw) [inline js/jquery.validation.js]: How about an svg image to start with in the images folder.

**Comment by nicky-johnstone** [comment]
That's an SVG icon added for the close button on message areas.

I've set it to 16px but not sure if it's too big

image

At smaller sizes it's off-center. Here it is at 10px:

image

**Comment by bvw** [comment]
@achisholm could you comment on this please.

**Comment by achisholm** [comment]
You could try adding margin/ padding to the top or if that messes up the text position, try adding line-height, often matching the text gets it about right.

Failing that you can nudge it down with something like `position:relative: top: 10px`

**Comment by achisholm** [comment]
I think the smaller one is better

**Comment by nicky-johnstone** [comment]
Updated.

image

**Pull request merged**